csp/models.py

```diff
@@ -49,25 +48,24 @@ def strip_uri(cls, uri: str) -> str:
         """
         return strip_query(uri)[:200] if uri else ""
 
-    @root_validator
-    def validate_directives(cls, values: PydanticValues) -> PydanticValues:
+    @model_validator(mode="after")
+    def validate_directives(self) -> ReportData:
```

Comment:
Using the `model_validator` with mode `before` would allow us to get the values as a dictionary, similar to what was happening with the deprecated `root_validator`.  However, it seems like using `before` with the `model_validator` ignores the field's `alias` parameter, which means that if users are passing `effective-directive`, that's what we get on the values dictionary.

I'm not sure if this Pydantic change was intentional or not, since `before` validators are supposed to validate before any parsing is done on the fields, but it seems to me like we should be able to access the original field by the field's name, not its alias.